docs/contributing.rst

```diff
@@ -2,12 +2,680 @@
+guide to complete this step.
+
+.. attention::
+    The basic steps for GitHub/Git configuration, which can be learned from the
+    linked guides here, need to be performed **before** contributing to UXarray!
+
+2.2. Python Environment Setup
+-----------------------------
+
+Before starting any Python or documentation development, you’ll need to
+create a Python environment along with a package manager. When we use the
+term “Python environment” here, we are referring to the Python programming
+language. Because there are so many Python packages available,
+maintaining interoperability between them is a huge challenge. To overcome
+some of these difficulties, we strongly recommend the use of `Anaconda
+<https://www.anaconda.com/download/>`_ or `Miniconda
+<https://conda.io/miniconda.html/>`_ as a package manager to manage your
+Python ecosystem. These package managers allow you to create a separate,
+custom Python environment for each specific Python set of tools. Yes, this
+unfortunately results in multiple copies of Python on your system, but it
+greatly reduces breaking toolchains whenever a change is made to your Python
+environment (and is more reliable than any other solution we’ve encountered).
+Also, the use of Anaconda/Miniconda is standard practice for working with
+Python in general, not simply for using UXarray.
+
+To configure your Python environment:
+
+1. Install either `Anaconda <https://www.anaconda.com/download/>`_ or
+`Miniconda <https://conda.io/miniconda.html/>`_.
+
+.. tip::
+    We recommend Miniconda as it does not unnecessarily install many
+    packages that will not be needed for UXarray development.
+
+2. Make sure your conda is up to date by running this command from the
+terminal::
+
+    $ conda update conda
+
+.. warning::
+    Don’t type the $ character. This simply indicates the command line prompt.
+
+At this point, you will have a current version of conda available on your
+system. Before using your Python environment to work on UXarray development,
+you’ll need to perform additional steps that are going to be described in
+the next section.
+
+3. How UXarray Uses Git/GitHub
+==============================
+
+UXarray uses the `GitHub Flow
+<https://docs.github.com/en/get-started/quickstart/github-flow>`_ model
+for its workflow. UXarray also uses an automated formatter, which is
+described in `3.4. Install and Setup Pre-commit Hooks`_ in detail, on
+all commits in local development environment in order to ensure code
+formatting. This changes the normal workflow slightly, so in order to
+avoid any confusions, follow these steps:
+
+3.1. Select An Issue to Work on
+-------------------------------
+
+Virtually any work should be addressing an issue. There are a few options to
+select an issue to work on:
+
+1. First, check the existing `UXarray issues
+<https://github.com/UXARRAY/uxarray/issues>`_. These issues might have been
+created from either the
+* `UXarray draft API
```

Comment:
The markdown format seems to be off here

Reply:
Done; thanks for catching this!